### ISSUE TITLE
fully_type_itabs: False positives on BDEF derived tables

### ISSUE THREAD

**Issue by agoeb** [abaplint/abaplint#3547]
The following statement raises a finding from the fully_type_itabs rule:

```
REPORT zfoo.
DATA foo TYPE TABLE FOR CREATE zr_foo.
```

The message is: `/zfoo.prog.abap[2, 10]: Specify table key(fully_type_itabs)` 
(verified on playground with most recent version)

This statement defines a [BDEF Derived Type](https://help.sap.com/doc/abapdocu_cp_index_htm/CLOUD/en-US/ABAPTYPE_TABLE_FOR.html), which does not accept defining table keys like classic itabs.

This message to be caused by `TYPE TABLE FOR` being recognized as `Expressions.TypeTable`. Not sure whether it would be better to fix this in the expression recognizer or in the rule itself.

**Comment by larshp** [comment]
@agoeb try again, latest version

**Comment by agoeb** [comment]
Thanks a lot!
